File: db/src/test/java/onl/ycode/stormify/TestLogger.java
package onl.ycode.stormify;

import onl.ycode.logger.Logger;
import onl.ycode.logger.SilentLogger;
import onl.ycode.logger.WatchLogger;

public class TestLogger implements WatchLogger.Watcher {

    private final Logger origLogger = StormifyManager.stormify().getLogger();
    private final StringBuilder buffer = new StringBuilder();

    // Constructor to set up the logger
    public TestLogger() {
        // Set the Stormify logger to use this TestLogger
        StormifyManager.stormify().setLogger(new WatchLogger(new SilentLogger(), this));
    }

    // Implementation of the WatchLogger.Watcher interface
    @Override
    public void watch(String level, String message, Throwable throwable) {
        buffer.append(message).append("\n");
    }

    public String get() {
        try {
            return buffer.toString();
        } finally {
            buffer.setLength(0); // Clear the buffer after returning the log
        }
    }

    // Method to restore the original logger
    public void close() {
        StormifyManager.stormify().setLogger(origLogger);
    }
}

File: kotlin/src/test/kotlin/onl/ycode/stormify/TestLogger.kt
package onl.ycode.stormify

import onl.ycode.logger.SilentLogger
import onl.ycode.logger.WatchLogger
import onl.ycode.stormify.StormifyManager.stormify

class TestLogger : WatchLogger.Watcher {
    private val origLogger = stormify().logger

    init {
        stormify().logger = WatchLogger(SilentLogger(), this)
    }

    private val buffer = StringBuilder()
    override fun watch(level: String, message: String, throwable: Throwable?) {
        buffer.append("$message\n")
    }

    operator fun invoke() = buffer.toString().also {
        buffer.clear()
    }

    fun close() {
        stormify().logger = origLogger
    }
}